phsb5321/aca.so
# Chore/services refactor

## Chore/services refactor: Split large service classes into smaller, more focused services

This PR refactors the codebase to break large service classes into smaller unit-sized services, improving code maintainability and organization. The changes include the following:

- Separation of service classes into individual files for each action, organized by entity (e.g., authors, books, people, and readers).
- Grouping of related services into subdirectories.
- Updating import statements and references to reflect the new file organization.

This refactoring effort ensures that each service class has a single responsibility, making it easier to understand and modify the codebase as needed.

### Changes

- Separated the large `people_services.py`, `books_services.py`, `authors_services.py`, and `readers_services.py` classes into smaller, more focused services.
- Grouped the individual service files into subdirectories: `authors_services`, `books_services`, `people_services`, and `readers_services`.
- Updated import statements and references throughout the codebase to reflect the new file organization.

### Testing

To test the refactored codebase, follow these steps:

1. Clone this repository and navigate to the root directory.
2. Install the required dependencies by running `poetry install`.
3. Start the Gremlin server by running `docker-compose up -d`.
4. Create a new virtual environment and activate it.
5. Start the API by running `uvicorn app.main:app`.
6. Use a tool like `curl`, `httpie`, or `Postman` to make requests to the API endpoints.
7. Verify that the responses match the expected results.
8. Run tests by executing the command `pytest` in the `tests` directory.
9. Verify that all tests pass.

Please review the changes and provide any feedback or approval. Thank you!
